test/integration/schemagic.spec.js

```diff
@@ -107,6 +106,98 @@ describe('/source/schemagic with valid example schemas', function () {
 				});
 			});
 		});
+
+		describe('validating against test1 schema that has valid foreign key value, ' +
+			'foreignKey defined with dot notation and belongs to the schema, foreignKey:true - callback', function(){
+			var result;
+			before(function(done){
+				var doc =  {
+					a: 1,
+					testForeignKey2: 12 //this is valid in 'test.testForeignKey2' foreign key check
+				};
+				var options = {foreignKey:true};
+				return schemagic.test.validate(doc, options, saveResult);
+
+				function saveResult(err, res){
+					if(err){
+						return done(err);
+					}
+					result = res;
+					return done();
+				}
+			});
+
+			it('should return result without error', function(){
```

Comment:
@ebdrup please don't blame me, i remember that i promised to not write comments about whitespaces, but this time @bubenshchykov asked me to do this)
`it('should return result without error', function()<wouldBeNiceToHaveSpaceHere>{`